src/controllers/auth.controller.js

```diff
@@ -0,0 +1,43 @@
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import User from '../models/user.js';
+import { RefreshTokenSchema } from '../models/refreshToken.js';
+import handleResponse from './handleResponse.js';
+
+dotenv.config();
+
+const AuthHandler = async (req, res) => {
+	const user = await User.findOneBy({
+		email: req.body.email,
+	});
+	if (!user) {
+		handleResponse(res, 404, { message: res.__('user does not exist') });
+	} else {
+		try {
+			if (await bcrypt.compare(req.body.password, user.password)) {
+				const accessToken = jwt.sign(
+					{ id: user.id },
+					process.env.ACCESS_TOKEN_SECRET,
+					{ expiresIn: '1h' }
+				);
+				const refreshToken = await RefreshTokenSchema.createToken(user);
+				res.cookie('jwt', refreshToken, {
+					httpOnly: true,
+					maxAge: 60 * 60 * 60 * 24 * 7,
+				});
+				handleResponse(res, 200, { access_token: accessToken });
```

Comment:
Apart from the token that can't be decrypted on the frontend. send back some information regarding the user. Those information include:
- First name (required)
- Role (required for navigation - frontend)
- Last name (optional)
- Email (optional)